src/features/transfer/TransferTokenForm.tsx

```diff
@@ -53,14 +56,14 @@ export function TransferTokenForm({ tokenRoutes }: { tokenRoutes: RoutesMap }) {
   const queryClient = useQueryClient();
   const { address: accountAddress } = useAccount();
   const validate = (values: TransferFormValues) =>
-    validateFormValues(values, queryClient, accountAddress);
+    validateFormValues(values, queryClient, accountAddress, isERC721);
 
   const onDoneTransactions = () => {
     setIsReview(false);
     // Consider clearing form inputs here
   };
-  const { triggerTransactions } = useTokenTransfer(onDoneTransactions);
-
+  const tokenTransaction = useTokenTransfer(onDoneTransactions);
+  const ERC721Transaction = useERC721Transfer(onDoneTransactions);
```

Comment:
Why make two separate hooks for this instead of having `useTokenTransfer` choose the right type of transfer based on the token type?